### ISSUE TITLE
Multiple Classification

### ISSUE THREAD

**Issue by AkSino** [triton-inference-server/server#4909]
**Is your feature request related to a problem? Please describe.**

Hello, I am currently using a token classification model and I would like the classification extension to return the label associated to each token. 

Maybe I am doing something wrong or maybe it's not possible but here is my problem:

_config.pbtxt (output)_
```pbtxt
...
output [
    {
        name: "logits"
        data_type: TYPE_FP32
        dims: [
             1,
            -1,
            -1
        ]
        label_filename: "labels.txt"
    },
    {
        name: "boxes"
        data_type: TYPE_INT64
        dims: [ -1, 4]
    }
]
```

_labels.txt_
```txt
labels_1
labels_2
...
labels_23
```

_client.py_
```python
# Initialize Triton Client
tritonclient_all = tritonclient.http.InferenceServerClient(url=URL, verbose=False)
query_image = tritonclient.http.InferInput(name='image', shape= (1684,1023,3 ), datatype="UINT8")
query_words = tritonclient.http.InferInput(name='words', shape=(468,), datatype="BYTES")
query_boxes = tritonclient.http.InferInput(name='boxes', shape=(468, 4), datatype="INT64")
query_labels = tritonclient.http.InferInput(name='labels', shape=(468,), datatype="INT64")

query_image.set_data_from_numpy(image)
query_words.set_data_from_numpy(encoded_words)
query_boxes.set_data_from_numpy(boxes)
query_labels.set_data_from_numpy(ner_tags)

output_logits = tritonclient.http.InferRequestedOutput("logits", binary_data=False, class_count=1)
output_boxes = tritonclient.http.InferRequestedOutput("boxes", binary_data=False)
response = tritonclient_all.infer(
        MODEL_NAME, model_version=MODEL_VERSION, inputs=[query_image, query_words, query_boxes, query_labels], outputs=[output_logits, output_boxes]
    )
classification=response.as_numpy('logits')
```

Notes: 

_logits shape=(1,1274,23) --> (batch_size, nb_tokens, nb_classes)_
```python
[[[ 3.2473, -0.9262, -0.8753,  ..., -0.3751, -0.5838, -0.5293],
[ 8.2308, -0.9632, -1.1002,  ..., -1.0459, -0.7460, -1.2538],
[ 8.1926, -0.9231, -1.0982,  ..., -1.0271, -0.7852, -1.2739],
...,
[ 6.7117, -1.1195, -1.1662,  ..., -1.2212, -0.4493, -1.4373],
[ 7.6300, -1.0269, -1.1790,  ..., -1.0924, -0.5879, -1.4987],
[ 8.1188, -0.9041, -1.1706,  ..., -1.0296, -0.7105, -1.2952]]]
```

And I got

_output_
`['8.361367:7912']`

**Describe the solution you'd like**

I would like to have a classification for each token:

_output_
`[['3.2473:0:labels_1'], ['8.2308:0:labels_1'], ..., ['8.361367:5:labels6'], ['6.6592:15:labels_16']]`

with a length of 1274 (nb_tokens)

Is it possible? If so, do you see anything I am doing wrong?

Thanks and have a good day

**Comment by jbkyang-nvi** [comment]
Hi @AkSino can you also share your input model config? 
cc: @tanmayv25 since I could not have multiple classifications on the [model configuration documentation[(https://github.com/triton-inference-server/server/blob/main/docs/user_guide/model_configuration.md)

**Comment by AkSino** [comment]
Hello @jbkyang-nvi 
Oh sure I can 

Here my complete config:

```
name: "layout_bev_1"
backend: "python"
version_policy: { all { }}
input [
{
        name: "image"
        data_type: TYPE_UINT8
        dims: [-1, -1, 3] 
},
{
        name: "words"
        data_type: TYPE_STRING
        dims: [ -1] 
},
{
        name: "boxes"
        data_type: TYPE_INT64
        dims: [ -1, 4] 
},
{
        name: "labels"
        data_type: TYPE_INT64
        dims: [ -1] 
}
]
output [
    {
        name: "logits"
        data_type: TYPE_FP32
        dims: [
             1,
            -1,
            -1
        ]
        label_filename: "labels.txt"
    },
    {
        name: "boxes"
        data_type: TYPE_INT64
        dims: [ -1, 4]
    }
]
```

Thanks

**Comment by tanmayv25** [comment]
> I would like to have a classification for each token:
> 
> output
> [['3.2473:0:labels_1'], ['8.2308:0:labels_1'], ..., ['8.361367:5:labels6'], ['6.6592:15:labels_16']]
> 
> with a length of 1274 (nb_tokens)

@AkSino  If you need the output for each tokens then you should provide class_count=1274.

**Comment by tanmayv25** [comment]
Read more about classification extension here: https://github.com/triton-inference-server/server/blob/c2a6ce18e212c4db6f1abd6435a077c952182558/docs/protocol/extension_classification.md

**Comment by tanmayv25** [comment]
@AkSino  Do you have 7912 entries in your labels.txt?

**Comment by AkSino** [comment]
Hello ! 

Thanks, it seems that I miss understand `class_count`, I'll try this.
@tanmayv25 No I have only 23 labels why?

**Comment by tanmayv25** [comment]
> @tanmayv25 No I have only 23 labels why?

From your output it looks like the index with max value was `7912`. Because there are only 23 labels, you can expect the labels suffix only for the indices within 1-23.

**Comment by tanmayv25** [comment]
> The BYTES output with its float:<count> format that would otherwise come out when class_count=61 and query_params=None looks very confusing to me.

class_count is triton-specific protocol extension. In order to minimize the data overhead, it only sends the top class_count many classes in format <value>:<index>:<optional_label>. In order to get complete output you are expected to provide class_count=0 as above.
